src/betterproto/__init__.py

```diff
@@ -495,7 +590,35 @@ class ParsedField:
     raw: bytes
 
 
-def parse_fields(value: bytes) -> Generator[ParsedField, None, None]:
+def load_fields(stream: BinaryIO) -> Iterator[ParsedField]:
+    while True:
+        try:
+            num_wire, raw = load_varint(stream)
+        except EOFError:
+            return
+        number = num_wire >> 3
+        wire_type = num_wire & 0x7
+
+        decoded: Any = None
+        if wire_type == WIRE_VARINT:
+            decoded, r = load_varint(stream)
+            raw += r
+        elif wire_type == WIRE_FIXED_64:
+            decoded = stream.read(8)
+            raw += decoded
+        elif wire_type == WIRE_LEN_DELIM:
+            length, r = load_varint(stream)
+            decoded = stream.read(length)
+            raw += r
+            raw += decoded
+        elif wire_type == WIRE_FIXED_32:
+            decoded = stream.read(4)
+            raw += decoded
+
+        yield ParsedField(number=number, wire_type=wire_type, value=decoded, raw=raw)
+
+
+def parse_fields(value: bytes) -> Iterator[ParsedField]:
```

Comment:
```suggestion
def parse_fields(value: bytes) -> Generator[ParsedField, None, None]:
```
Not really sure why these were touched

Reply:
Reverted